### DISCUSSION TITLE
How to use jsc.transform.react.refresh correctly

### DISCUSSION THREAD

**Discussion by HeleneGrini** [swc-project/swc#1855] [Q&A]
Hello, 

I want to enable fast refresh, but I cant seem to get it right.  Here's my .swcrc file 

```
[
  {
    "test": ".tsx?$",
    "jsc": {
      "parser": {
        "syntax": "typescript",
        "tsx": true,
        "decorators": false,
        "dynamicImport": true
      },
      "transform": {
        "react": {
          "refresh": true
        }
      }
    }
  },
   {
    "test": ".jsx?$",
    "jsc": {
      "parser": {
        "syntax": "ecmascript",
        "jsx": true,
        "decorators": false,
        "dynamicImport": true
      },
      "transform": {
        "react": {
          "refresh": true
        }
      }
    }
  }
]
```

When I added the `refersh` the page does not reload, but it does not update either. Am I missisng something? 

I couldn't find any examples in the docs, and if "refresh" is supposed to be an object with more options like this interface in the docs, then what do these options mean? 

```
interface ReactRefreshConfig {
  refreshReg: String;
  refreshSig: String;
  emitFullSignatures: boolean;
}
```
Thanks!

**Comment by kdy1** [comment]
How did you invoked swc?
I'm not sure about this, but I think you should use something like webpack with swc-loader.

**Reply by HeleneGrini** [reply]
Yes, I'm using webpack with swc-loader, have been for months. Works smootly, it's just the fast refresh not working as expected

**Reply by kdy1** [reply]
Oh, I checked the `swc-loader`, and it seems like it does not use react fast refresh correctly.

**Reply by HeleneGrini** [reply]
I see! Are you also maintaining the loader?

**Reply by kdy1** [reply]
Yes, it's maintained by me

**Reply by HeleneGrini** [reply]
When is it likely to have a fix for this issue?

**Reply by kdy1** [reply]
I don't have any estimate

**Reply by fforres** [reply]
Hey @kdy1 have you worked on it lately? I can help, but I'm a bit lost on the details. 
You have any idea on how should I go about this? Any guidance on how to do so, or if you have any info/reading etc would be great :)

**Reply by kdy1** [reply]
I'm not sure because I don't know well about how react fast refresh works. Some users are already using fast refresh with swc, but I'm not sure about how.
